numpy/core/records.py

```diff
@@ -661,6 +661,8 @@ def fromrecords(recList, dtype=None, shape=None, formats=None, names=None,
     """
 
     if formats is None and dtype is None:  # slower
+        if len(recList) == 0:
+            return record([])
```

Comment:
`record([])` seems broken to me - I'd expect it to return an object with `dtype.names == ()`, but it returns a plain `void`

Reply:
good catch, tested and fixed in 44106be